### PR TITLE
Refactor person to contact in diagrams, fixed previous refactor's NPE

### PR DESCRIPTION
**Pull request by McNaBry**
- Fix null pointer exception caused by refactoring the UI classes. Since the JavaFX id was not updated properly, this resulted in the NPE at runtime.
- Included some updates left out in the previous refactor for diagrams in documentation.

### CODE REVIEW

File: src/main/resources/view/ContactListPanel.fxml
Comment (CJ-Lee01) [inline src/main/resources/view/ContactListPanel.fxml:1]: Looks good

**Review by CJ-Lee01 — APPROVED** [review]
Looks good to me.
Are the changes in the docs necessary for the bug fix?

**Comment by wxwern** [comment]
I have reworded the PR to match what's committed for less confusion. I would say we shouldn't be putting both the bug fix from refactor and other refactoring commits in the same PR, but since they're related to the same refactor I think I can let it slide.

**Review by tanshiyu1999 — APPROVED** [review]
LGTM

**Pull request merged**